bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/firmware/Firmware.java

```diff
@@ -82,6 +93,8 @@ private Firmware(Builder builder) {
         this.onlineChangelog = builder.onlineChangelog;
         this.inputStream = builder.inputStream;
         this.md5Hash = builder.md5Hash;
+        this.properties = Collections
+                .unmodifiableMap(builder.properties != null ? builder.properties : new HashMap<>());
```

Comment:
Maybe you can use `Collections.emptyMap()` here instead of (unused) `new HashMap()`.